src/main/java/org/mule/extension/sftp/internal/connection/SftpClient.java

```diff
@@ -6,108 +6,102 @@
+import org.mule.extension.sftp.internal.error.FileError;
+import org.mule.extension.sftp.internal.proxy.http.HttpClientConnector;
+import org.mule.extension.sftp.internal.proxy.socks5.Socks5ClientConnector;
 import org.mule.runtime.api.connection.ConnectionException;
 import org.mule.runtime.api.exception.MuleRuntimeException;
 
-import com.jcraft.jsch.Channel;
-import com.jcraft.jsch.ChannelSftp;
-import com.jcraft.jsch.JSch;
-import com.jcraft.jsch.JSchException;
-import com.jcraft.jsch.Proxy;
-import com.jcraft.jsch.ProxyHTTP;
-import com.jcraft.jsch.ProxySOCKS4;
-import com.jcraft.jsch.ProxySOCKS5;
-import com.jcraft.jsch.Session;
-import com.jcraft.jsch.SftpException;
-
 import java.io.File;
 import java.io.IOException;
 import java.io.InputStream;
 import java.io.OutputStream;
+import java.net.InetSocketAddress;
 import java.net.URI;
+import java.nio.file.Paths;
+import java.security.GeneralSecurityException;
+import java.security.KeyPair;
+import java.util.Collection;
 import java.util.List;
-import java.util.Properties;
-import java.util.function.Supplier;
 
+import org.apache.sshd.client.SshClient;
+import org.apache.sshd.client.session.ClientSession;
+import org.apache.sshd.common.config.keys.FilePasswordProvider;
+import org.apache.sshd.common.config.keys.loader.KeyPairResourceLoader;
+import org.apache.sshd.common.keyprovider.KeyIdentityProvider;
+import org.apache.sshd.common.util.security.SecurityUtils;
+import org.apache.sshd.sftp.client.SftpClient.OpenMode;
+import org.apache.sshd.sftp.common.SftpConstants;
+import org.apache.sshd.sftp.common.SftpException;
 import org.slf4j.Logger;
-import org.slf4j.LoggerFactory;
 
 /**
- * Wrapper around jsch sftp library which provides access to basic sftp commands.
+ * Wrapper around apache sshd library which provides access to basic sftp commands.
  *
  * @since 1.0
  */
 public class SftpClient {
 
-  private static Logger LOGGER = LoggerFactory.getLogger(SftpClient.class);
-
-  private static SftpLogger JSCH_LOGGER = new SftpLogger();
-
-  public static final String CHANNEL_SFTP = "sftp";
-  public static final String STRICT_HOST_KEY_CHECKING = "StrictHostKeyChecking";
-  public static final String PREFERRED_AUTHENTICATION_METHODS = "PreferredAuthentications";
+  private static final Logger LOGGER = getLogger(SftpClient.class);
+  protected static final OpenMode[] CREATE_MODES = {OpenMode.Write, OpenMode.Create};
+  protected static final OpenMode[] APPEND_MODES = {OpenMode.Write, OpenMode.Append};
 
-  static {
-    JSch.setLogger(JSCH_LOGGER);
-  }
-
-  private ChannelSftp sftp;
-  private JSch jsch;
-  private Session session;
+  private final SshClient client = SshClient.setUpDefaultClient();
+  private org.apache.sshd.sftp.client.SftpClient sftp;
+  private ClientSession session;
   private final String host;
   private int port = 22;
   private String password;
   private String identityFile;
   private String passphrase;
   private String knownHostsFile;
```

Comment:
Why the KnownHostFile feature is commented?
Is it not supported by the newer implementation?